packages/babel-plugin-transform-runtime/src/helpers.js

```diff
@@ -32,6 +32,57 @@ export function hasMinVersion(minVersion, runtimeVersion) {
   );
 }
 
+export function verifyRuntimeVersion(
+  options,
+  moduleName,
+  runtimeVersion,
+  actualRuntimeVersion,
+) {
+  if (!actualRuntimeVersion) return;
+
+  const minActualVersion = semver.minVersion(actualRuntimeVersion).version;
+  const minVersion = semver.minVersion(runtimeVersion).version;
+
+  if (minActualVersion === minVersion) return;
```

Comment:
Treating semver ranges as anything other than ranges is asking for trouble I think. Avoiding logic like this is why I lean toward `intersects` for things in https://github.com/babel/babel/blob/f0c2364d0177908da4fab336fb2b1ad1b1a082d7/packages/babel-core/src/transformation/file/file.js#L191.

What we really want to know in this context is whether every possible version in `actualRuntimeVersion` is contained within `runtimeVersion`, I think? As in, every possible version of the runtime that could be installed by this package.json is valid given the semver range passed to the Babel plugin.

This current logic would pass if for instance we had
```
runtimeVersion: "^7.2.0",
actualRuntimeVersion: "~7.2.0"
```
since the minimum for both is `7.2.0`.

I'm not sure if the semver module actually exposes and API for doing the comparison we need to do here, but we can certainly check.